src/main/java/net/dries007/tfc/objects/blocks/BlockFlowerPotTFC.java

```diff
@@ -0,0 +1,195 @@
+    {
+        return false;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockFaceShape getBlockFaceShape(IBlockAccess worldIn, IBlockState state, BlockPos pos, EnumFacing face)
+    {
+        return face == EnumFacing.DOWN ? super.getBlockFaceShape(worldIn, state, pos, face) : BlockFaceShape.UNDEFINED;
+    }
+
+    @Override
+    public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, EntityPlayer player, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
+    {
+        if (!world.isRemote)
+        {
+            TEFlowerPotTFC te = Helpers.getTE(world, pos, TEFlowerPotTFC.class);
+            if (te != null && te.isEmpty())
+            {
+                ItemStack held = player.getHeldItem(hand);
+                IBlockState putState = Block.getBlockFromItem(held.getItem()).getDefaultState();
+                // allow addons to skip the plant check optionally. None of our plants use this name so this statement will always return true in vanilla tfc
+                if (!OreDictionaryHelper.doesStackMatchOre(held, "canBePotted"))
+                {
+                    putState = tryMakePlantState(putState, true);
+                }
+                te.setState(putState);
+                if (!player.isCreative() && putState.getBlock() != Blocks.AIR)
+                    held.shrink(1);
+            }
+        }
+        return true;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public boolean canPlaceBlockAt(World worldIn, BlockPos pos)
+    {
+        IBlockState downState = worldIn.getBlockState(pos.down());
+        return super.canPlaceBlockAt(worldIn, pos) && (downState.isTopSolid() || downState.getBlockFaceShape(worldIn, pos.down(), EnumFacing.UP) == BlockFaceShape.SOLID);
+    }
+
+    @Override
+    public void onBlockHarvested(World world, BlockPos pos, IBlockState state, EntityPlayer player)
+    {
+        TEFlowerPotTFC te = Helpers.getTE(world, pos, TEFlowerPotTFC.class);
+        if (te != null && !player.isCreative())
+        {
+            te.dump();
+        }
+    }
+
+    @Override
+    public ItemStack getPickBlock(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player)
+    {
+        TEFlowerPotTFC te = Helpers.getTE(world, pos, TEFlowerPotTFC.class);
+        if (te != null)
+        {
+            return new ItemStack(Item.getItemFromBlock(te.state.getBlock()));
+        }
+        return super.getPickBlock(state, target, world, pos, player);
+    }
+
+    @Override
+    public BlockRenderLayer getRenderLayer()
+    {
+        return BlockRenderLayer.CUTOUT;
+    }
+
+    @Override
+    protected BlockStateContainer createBlockState()
+    {
+        IProperty[] listedProperties = new IProperty[] {};
+        IUnlistedProperty[] unlistedProperties = new IUnlistedProperty[] {FLOWER};
+        return new ExtendedBlockState(this, listedProperties, unlistedProperties);
```

Comment:
There's a builder pattern for this: `new BlockStateContainer.Builder(this).add(FLOWER).build()`